### PR TITLE
Feat/delete log file

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | follow #12636 

(also contains a fix for broken breadcrumbs following the refactor by @cedric-anne )

image

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: front/logviewer.php [hunk 1]
```diff
@@ -50,16 +50,20 @@
 if (($_GET['action'] ?? '') === 'download') {
     $logparser = new LogParser();
     $logparser->download($filepath);
-} elseif (($_POST['action'] ?? '') === 'empty') {
+} elseif (($_REQUEST['action'] ?? '') === 'empty') {
```
Comment (cedric-anne) [inline front/logviewer.php:53]: This is a writing operation, CSRF token have to be validated.
```suggestion
} elseif (($_POST['action'] ?? '') === 'empty') {
```

File: front/logviewer.php [hunk 2]
```diff
@@ -50,16 +50,20 @@
...
     $logparser = new LogParser();
     $logparser->empty($filepath);
     Html::back();
+} elseif (($_REQUEST['action'] ?? '') === 'delete') {
```
Comment (cedric-anne) [inline front/logviewer.php:57]: This is a writing operation, CSRF token have to be validated
```suggestion
} elseif (($_POST['action'] ?? '') === 'delete') {
```

File: templates/pages/admin/logs_list.html.twig [hunk 3]
```diff
@@ -106,8 +107,40 @@
                                       title="{{ __("date of last modification") }}">
                                     {{ log['datemod']|formatted_datetime }}
                                 </span>
+
+                                {% if can_clear or can_delete %}
+                                    <span>
```
Comment (cedric-anne) [inline templates/pages/admin/logs_list.html.twig:112]:
```suggestion
                                    <form action="{{ path('/front/logviewer.php') }}" method="post" class="d-inline">
                                    <input type="hidden" name="filepath" value="{{ filepath }}">
                                    <input type="hidden" name="_glpi_csrf_token" value="{{ csrf_token() }}">
                                    <span>
```

File: templates/pages/admin/logs_list.html.twig [hunk 4]
```diff
@@ -106,8 +107,40 @@
...
+                                        </a>
+                                        <ul class="dropdown-menu">
+                                            {% if can_clear %}
+                                                <li>
+                                                    <a class="dropdown-item"
+                                                       href="{{ path('/front/logviewer.php?filepath=' ~ log['filepath']|url_encode ~ '&action=empty') }}"
+                                                       onclick="return confirm('{{ __("Are you sure you want to clear this file?") }}')">
+                                                        <i class="ti ti-file-off"></i>
+                                                        <span>{{ __("Empty file") }}</span>
+                                                    </a>
```
Comment (cedric-anne) [inline templates/pages/admin/logs_list.html.twig:126]:
```suggestion
                                                    <button type="submit" name="action" value="empty"
                                                                 class="dropdown-item"
                                                                 title="{{ __("Empty file") }}"
                                                                 onclick="return confirm('{{ __("Are you sure you want to clear this file?") }}')"
                                                                 data-bs-toggle="tooltip" data-bs-placement="bottom">
                                                        <i class="ti ti-file-off"></i>
                                                        <span>{{ __("Empty file") }}</span>
                                                    </button>
```

File: templates/pages/admin/logs_list.html.twig [hunk 5]
```diff
@@ -106,8 +107,40 @@
...
+                                                </li>
+                                            {% endif %}
+                                            {% if can_delete %}
+                                                <li>
+                                                    <a class="dropdown-item"
+                                                       href="{{ path('/front/logviewer.php?filepath=' ~ log['filepath']|url_encode ~ '&action=delete') }}"
+                                                       onclick="return confirm('{{ __("Are you sure you want to delete this file?") }}')">
+                                                        <i class="ti ti-trash-x"></i>
+                                                        <span>{{ __("Delete file") }}</span>
+                                                    </a>
```
Comment (cedric-anne) [inline templates/pages/admin/logs_list.html.twig:136]:
```suggestion
                                                    <button type="submit" name="action" value="delete"
                                                                 class="dropdown-item"
                                                                 title="{{ __("Empty file") }}"
                                                                 onclick="return confirm('{{ __("Are you sure you want to delete this file?") }}')"
                                                                 data-bs-toggle="tooltip" data-bs-placement="bottom">
                                                        <i class="ti ti-trash-x"></i>
                                                        <span>{{ __("Delete file") }}</span>
                                                    </button>
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**